src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -442,7 +442,7 @@ MatterThermostatClusterServerPreAttributeChangedCallback(const app::ConcreteAttr
         }
         auto RequestedSystemMode = static_cast<SystemModeEnum>(*value);
         if (ControlSequenceOfOperation > ControlSequenceOfOperationEnum::kCoolingAndHeatingWithReheat ||
-            RequestedSystemMode > SystemModeEnum::kFanOnly)
+            EnsureKnownEnumValue(RequestedSystemMode) != SystemModeEnum::kUnknownValue)
```

Comment:
This check seems backwards.  Was this tested?  Given that it fails to compile, I assume not....

Please add a test that would have caught the old (incorrect) code.